enterprise/frontend/src/embedding-sdk/store/reducer.ts

```diff
@@ -48,12 +55,27 @@ export const getOrRefreshSession = createAsyncThunk(
 
 export const refreshTokenAsync = createAsyncThunk(
   REFRESH_TOKEN,
-  async (url: string): Promise<EmbeddingSessionTokenState["token"]> => {
-    const response = await fetch(url, {
-      method: "GET",
-      credentials: "include",
-    });
-    return await response.json();
+  async (
+    url: string,
+    { getState },
+  ): Promise<EmbeddingSessionTokenState["token"]> => {
+    // The SDK user can provide a custom function to refresh the token.
+    const getRefreshToken =
+      getFetchRefreshTokenFn(getState() as SdkStoreState) ??
+      defaultGetRefreshTokenFn;
+
+    try {
+      return await getRefreshToken(url);
+    } catch (errorCause) {
+      // As this function can be supplied by the SDK user,
+      // we have to handle possible errors in refreshing the token.
+      const error = new Error(t`failed to refresh the auth token`);
+      error.cause = errorCause;
+
+      setLoginStatus({ status: "error", error });
+
+      return null;
+    }
```

Comment:
@oisincoveney I'm not sure if using `setLoginStatus` makes sense here - in this case this happens when we fail to refresh the authentication token, so I think it makes sense to set the login status to "error" here. As this function can be specified by the SDK user, there's always a possibility of error.